mantidproject/mantid
# fix crash on convFit when showing full function

**Description of work.**

This PR fixes a bug that caused a crash when you set the function as one of the InmutableConvolutionFunctions and then change it to show the full function.

**Report to:** Spencer Howells

**To test:**

1. Open Indirect data analysis
2. go to convtab
3. choose IsoRotDiff as the function
4. change to show full function and make sure it does not crash
5. try with DiffSphere and and DiffRotDiscreteCircle and amke sure they do not crash.

Fixes #31169

*This does not require release notes* because these functions were added in this release

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
